src/ecpool_worker.erl

```diff
@@ -211,8 +214,11 @@ code_change(_OldVsn, State, _Extra) ->
 %% Internal Functions
 %%--------------------------------------------------------------------
 
-connect(#state{mod = Mod, opts = Opts, id = Id}) ->
-    Mod:connect([{ecpool_worker_id, Id} | connopts(Opts, [])]).
+connect(#state{mod = Mod, opts = Opts, id = Id, initiated = Initiated,
+               connection_attempts = Attempts}) ->
+    ConOpts = [ {ecpool_worker_id, Id}, {initiated, Initiated}, {connection_attempts, Attempts}
```

Comment:
why pass down internal flags to callback ?
unknown options may cause unexpected behaviour in the callback

for testing ? maybe call sys:get_state to inspect the internal state.
or use https://github.com/kafka4beam/snabbkaffe

Reply:
The two flags are sent for the users benefit, it's necessary to allow the callbacks to make better connect behaviour decisions.

Reply:
in that case, maybe namespace them, e.g. `ecpool_is_retry_attempt` and  `ecpool_connection_attempt` ?

Reply:
or maybe this: `{ecpool_connect_type, initial | initial_retry | normal_retry}`